tests/units/Ticket.php

```diff
@@ -1077,4 +1077,174 @@ public function testUpdateFollowup() {
          ->variable['users_id']->isEqualTo($uid)
          ->variable['users_id_editor']->isEqualTo($uid2);
    }
+
+   protected function _testGetTimelinePosition($tlp, $tickets_id) {
+      foreach ($tlp as $users_name => $user) {
+         $auth = new \Auth();
+         $this->boolean((boolean)$auth->login($users_name, $user['pass'], true))->isTrue();
+         $uid = getItemByTypeName('User', $users_name, true);
+
+         // TicketFollowup
+         $fup = new \TicketFollowup();
+         $this->integer(
+            (int)$fup->add([
+               'tickets_id'   => $tickets_id,
+               'users_id'     => $uid,
+               'content'      => 'A simple followup'
+            ])
+         )->isGreaterThan(0);
+
+         $this->integer(
+            (int)$fup->fields['timeline_position']
+         )->isEqualTo($user['pos']);
+
+         // TicketTask
+         $task = new \TicketTask();
+         $this->integer(
+            (int)$task->add([
+               'tickets_id'   => $tickets_id,
+               'users_id'     => $uid,
+               'content'      => 'A simple Task'
+            ])
+         )->isGreaterThan(0);
+
+         $this->integer(
+            (int)$task->fields['timeline_position']
+         )->isEqualTo($user['pos']);
+
+         // Document and Document_Item
+         $doc = new \Document();
+         $this->integer(
+            (int)$doc->add([
+               'users_id'     => $uid,
+               'tickets_id'   => $tickets_id,
+               'name'         => 'A simple document object'
+            ])
+         )->isGreaterThan(0);
+
+         $doc_item = new \Document_Item();
+         $this->integer(
+            (int)$doc_item->add([
+               'users_id'      => $uid,
+               'items_id'      => $tickets_id,
+               'itemtype'      => 'Ticket',
+               'documents_id'  => $doc->getID()
+            ])
+         )->isGreaterThan(0);
+
+         $this->integer(
+            (int)$doc_item->fields['timeline_position']
+         )->isEqualTo($user['pos']);
+
+         // TicketValidation
+         $val = new \TicketValidation();
+         $this->integer(
+            (int)$val->add([
+               'tickets_id'   => $tickets_id,
+               'comment_submission'      => 'A simple validation',
+               'users_id_validate' => 5, // normal
+               'status' => 2
+            ])
+         )->isGreaterThan(0);
+
+         $this->integer(
+            (int)$val->fields['timeline_position']
+         )->isEqualTo($user['pos']);
+      }
+   }
+
+   protected function _testGetTimelinePositionSolution($tlp, $tickets_id) {
+      foreach ($tlp as $users_name => $user) {
+         $auth = new \Auth();
+         $this->boolean((boolean)$auth->login($users_name, $user['pass'], true))->isTrue();
```

Comment:
maybe factoring, DbTestCase::login (Ticket extends that class) could be enhanced to support username/password parameters

Reply:
DbTestCase::login doesn't permit login with 'glpi' user, but only with TU_USER

Reply:
that's exactly my point, i suggest a enhancement oh this function

Reply:
sorry I didn't understand, I agree, I'm going to extend it